rad/src/lib.rs

```diff
@@ -1526,6 +1572,58 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_try_data_request_http_get_non_ascii_header_key() {
+        let script_r = Value::Array(vec![]);
+        let packed_script_r = serde_cbor::to_vec(&script_r).unwrap();
+        let body = Vec::from(String::from(""));
+        let headers = vec![("ñ", "value")];
+        let headers = headers
+            .into_iter()
+            .map(|(a, b)| (a.to_string(), b.to_string()))
+            .collect();
+        let request = RADRequest {
+            time_lock: 0,
+            retrieve: vec![RADRetrieve {
+                kind: RADType::HttpGet,
+                url: String::from("http://127.0.0.1"),
```

Comment:
Hmmm I don't think this test can be run by default. For any context without an actual server at `127.0.0.1:80` it might fail with a TCP connection error instead of an invalid HTTP header error.

If that's right, my suggestion is to disable this test by default with the `#[ignore]` flag below `#[test]`. Then, when you want to run it on purpose, you only need to append `--ignored` to your `cargo test` command.

Reply:
All `test_try_data_request_http_*` tests, including this one, do run by default, because they're expecting some specific error to be thrown from the http lib (isahc) before an HTTP request is actually sent.